### ISSUE TITLE
Provide API to attach an accessible name to PRC `ProgressBar`

### ISSUE THREAD

**Issue by lesliecdubs** [primer/react#3605]
Because of `role: progressbar` AXE will break if no accessible name is provided (i.e.`aria-label` or `aria-labelledby`). I added an aria-label to all the examples but wonder if I should enforce it at a prop level or throw an error like we do witht he `children & progress`

_Originally posted by @kendallgassner in https://github.com/primer/react/issues/3517#issuecomment-1636402127_

**Comment by lesliecdubs** [comment]
Discussed at today's PRC maintainer's sync:
- seems like throwing a warning for this is the right path, but that won't pass CI in dotcom
- investigate function overloading to see if we could use the new `aria` attributes to opt into the correct props
